micro-rdk/src/common/board.rs

```diff
@@ -210,6 +228,16 @@ impl Board for FakeBoard {
         Err(BoardError::I2CBusNotFound(name))
     }
 
+    fn add_digital_interrupt_callback(
+        &mut self,
+        _pin: i32,
+        _intr_type: InterruptType,
+        _callback: Option<unsafe extern "C" fn(_: *mut core::ffi::c_void)>,
+        _arg: Option<*mut core::ffi::c_void>,
+    ) -> Result<(), BoardError> {
+        unimplemented!();
```

Comment:
we shouldn't call a function that will cause a panic even for a fakeboard implementation. The correct thing to do here is the return an error saying that the method is not supported